### DISCUSSION TITLE
Considering Windows 7 support

### DISCUSSION THREAD

**Discussion by Jan200101** [R2Northstar/NorthstarLauncher#668] [Ideas]
TL;DR
Windows 7 not supported, no more updates
old, doesn't receive newer software, cant use new winapi calls.
Drop? yay or nay?

----

Recently the topic of Northstar mixing and matching wide and normal width strings and characters came up and a discussion ensued what the correct way forward would be.

While on that subject Windows 7 came up.
Its been discontinued since 2020,
is not supported by a majority of modern applications (e.g. Firefox),
is no longer supported by Steam,
and has overall limited what we can use from the winapi or required workarounds for dependencies.

The question now is, should Northstar still support Windows 7?
pg has looked up that there are roughly 50 people using Northstar on Windows 7, though it is unclear how well Northstar truly works on Windows 7 at the moment (see https://github.com/R2Northstar/NorthstarLauncher/issues/612).

Relevant discussion: https://github.com/R2Northstar/NorthstarWiki/issues/261

**Comment by GeckoEidechse** [comment]
Use this comment to vote.

React with
- :+1: to **keep** Windows 7 support
- :-1: to **drop** Windows 7 support

**Reply by GeckoEidechse** [reply]
Fun fact, this vote used to be in favour of dropping Windows 7 support until someone removed their vote?

**Comment by GeckoEidechse** [comment]
My two cents on this is that we should only maintain Win7 support in cases where it's trivial to do so. If it's not trivial, it should not be given priority though I'm fine with taking PRs that maintain support ^^

**Reply by GeckoEidechse** [reply]
On that note, IMO, based on Steam stats Linux support should also have higher priority than Windows 7

**Reply by p0358** [reply]
Probably the most reasonable general way of going about it

**Comment by Jan200101** [comment]
Personally I am against Windows 7 support.

Its a platform left to die by its creator with little to no support by the community.
This makes it difficult to build for or test for without having hardware and software that works with it. I had recently ran into this myself when trying to run Northstar on Windows 7 on a T400p where I was constantly running into issues with nothing wanting to support the OS anymore.

It also doesn't help that supporting it would prevent us from adopting newer features the winapi provides such as UTF-8 support.

**Reply by p0358** [reply]
It does not prevent you from this. We use UTF-8 on -A version of APIs in TFORevive. It's optional. The worst that's gonna happen is that non-ASCII paths will break on Windows 7, but will work on Windows 10+. But you can fix your folder name on Windows 7 and then it still works. So it looks like the whole discussion arose from a misunderstanding?

**Comment by EM4Volts** [comment]
Windows 7 is 15 years old, has been discontinued, has no support anymore, is NOT supported by steam anymore (the main platform people own tf2 on, which could stop working there at any moment, which would make all the efforts to support it just poof away anyway).
Its 2024, if your pc cannot handle windows 10 (which becomes eol this year aswell) then it definitely cant handle tf2. im not saying ur pc sucks, dont play the game but people are definitely not keeping using win 7 for sane and good reasons.

**Reply by EM4Volts** [reply]
have to correct myself. win10 goes eol next year (not like that changes much about this discussion)

**Reply by p0358** [reply]
> steam anymore (the main platform people own tf2 on

That should need some stats to back that up, as it might not necessarily be true otherwise at all.

Both Origin and EA App explicitly support Windows 7, and they're the platforms you ultimately play the game on.

**Reply by EM4Volts** [reply]
> That should need some stats to back that up, as it might not necessarily be true otherwise at all.

here are ur 1 google search away facts:
https://help.steampowered.com/en/faqs/view/4784-4F2B-1321-800A

edit:
i dont think stats are needed to say that most people own the game on steam and not on the ea app.

**Reply by Jan200101** [reply]
I agree with p0, steam is the last platform to receive the game and thus the smallest amount of bought copies.

> Both Origin and EA App explicitly support Windows 7

I don't think this matters.
Origin is EOL (no matter how much you want to keep using it) and support for Windows 7 and 8 is considered second class for the EA App, with them being aware of a multitude of problems.

At the end I can only see explicit Windows 7 support to require us to adapt the project in multiple ways, such as migrating from libcurls's schannel backend to OpenSSL, or forking our dependencies to use older API calls.

Windows 7 has been released in 2009 with Microsoft ending mainstream support in 2015 and extended support in 2020.
Its been almost a decade since they ended support for end-users, we shouldn't continue to force it for the few people who are too lazy or ignorant to upgrade, even when it was free.

**Reply by EM4Volts** [reply]
i also want to just mention that if valve decides to kill off steam on windows 7 in the future all efforts will just break. sure you can say that about any os because they can kill off thoose aswell but right now they literally ended support for win 7.

EA app / Origin (i can promise you there arent many still using origin with some scuffed modifications just because they want origin) its just a matter of ur schizophrenic uncle named "electronic arts" to either fuck up or no longer care about win 7 support either.

update ur out of date os and then ur fine.

**Reply by p0358** [reply]
> i dont think stats are needed to say that most people own the game on steam and not on the ea app.

I have credible stats for Titanfall 1 that overwhelming majority is not playing with Steam, I suspect the situation isn't much different with Titanfall 2, so yes, but the other way around than you think. The game was on Origin for years before Steam, it's cheaper on EA's store, it has cd-keys there etc.

> i also want to just mention that if valve decides to kill off steam on windows 7 in the future all efforts will just break

They won't because EA App and Origin both support Windows 7 with no drop of support being announced, and the majority of people play there. There might even be workarounds for people to play the games they own through Steam without Steam.

> i can promise you there arent many still using origin with some scuffed modifications just because they want origin

Only 100k downloads, but it matters little if the new app also supports it.

> At the end I can only see explicit Windows 7 support to require us to adapt the project in multiple ways, such as migrating from libcurls's schannel backend to OpenSSL, or forking our dependencies to use older API calls.

Is there any other dependency that proved to be problematic in this area? As for schannel, it should still work perfectly fine as Windows 7 supported TLS 1.2, unless you're a schizo who wants to enforce 1.3 xd Root CAs should be fine unless someone didn't install all Windows Updates, but that would actually be the user's problem, and they'd have more breakage at that point so yeah.

In fact, I was testing using curl with schannel on Windows 7 recently and didn't run into any issues.

> Origin is EOL (no matter how much you want to keep using it) and support for Windows 7 and 8 is considered second class for the EA App, with them being aware of a multitude of problems.

Yeah, but at the end of the day both work. The matter should be revisited when both Origin is killed off and EA App drops Windows 7 support, making it thus impossible to play the game pretty much (except with some scuffed LSX proxy I guess lmao). Otherwise I don't see a reason to explicitly block off Windows 7 from playing, considering some work was put in to support it already (not much but still) and considering both the original game and platforms support it, just makes little sense to throw it out at this point yet, even though it's on the edge.

And I believe that day when it's dropped off by EA will come, just not there yet. Same as they had legacy Origin available for XP and Vista and killed it a few years later (in 2019 iirc). At that point it will be much more reasonable, as there won't be any realistically remaining players anymore + no new hardware will supporting booting it either for long time.

**Reply by GeckoEidechse** [reply]
I have the impression that we are basing this discussion on different views in regards to proportion of players on Steam vs Origin/EA-App ^^"

We can all agree that we should do whatever benefits the majority of players.

If we were go off of Steam players counts, than that would for example imply a higher focus on Linux users compared to Windows 7 (with Win10/11 still being top priority).

As we don't have statistics on the distribution of Steam vs EA App users on Northstar however, it's impossible to argue about the actual distribution.

We can at best only make guesses but those can turn out to be very much incorrect.

&nbsp;

Therefore I think it's instead best to just put the power of which platform to support into the hands of the individual contributor.

If someone is willing to go out of their way to make a PR that maintains compatibility with Windows 7, then I see close to no reason, not to merge it ^^

On the flipside, if a PR adds new useful functionality or fixes a crucial bug but in the process, breaks Win7 compatibility, then we will still merge it as the benefits to the many outweight those to the few.
Of course, someone else then stepping in and improving the change the add back Win7 support is still welcome.

Personally I'm focusing on Win10/11 and Linux but as we allow and respect (proper) reviews from all contributors, anyone that feels the need to ensure Win7 compatibility is more than encouraged to help out in that area <3

**Reply by EM4Volts** [reply]
> If someone is willing to go out of their way to make a PR that maintains compatibility with Windows 7, then I see close to no reason, not to merge it ^^

the issue is that this actively supports people that use an out of date system for no reason and gives them confirmation about their os still being totally fine.

it is end of life ALREADY, it is not the problem of any project now if a project no longer works on an old, eol system. and eol oses shouldnt be activly supported

**Comment by p0358** [comment]
Not surprised the usage shrunk in past two years since Northstar was released, given browsers and Steam now don't support it. Though I get why people still want to try it out (it's a simpler less bloated system with much better latency).

Now for this and the tiny population that plays it, maybe it's not worth keeping the support, but at the same time it feels wrong to outright remove support for something that used to work. The base game supports Windows 7 and Northstar generally does too. I think it would make sense to keep the support on a best-effort basis: no need to bother supporting extra features if it's not feasible, but compatibility shouldn't be outright broken by eg. statically importing WinAPI calls that aren't available on the older systems. I think it's a relatively tiny effort (just don't break what works for the most part) for this...

**Comment by EM4Volts** [comment]
Considering Win7 is even older now and we have even less people working on thinks I think this can finally be concluded as NOT supporting windows 7.

**Reply by p0358** [reply]
I am more inclined to agree at this point, the usage has dropped off since Steam ceased support and Origin got shut down. Now EA App seems to have silently dropped support too? Their cached DuckDuckGo site description from September mentions Windows 7 support, but the current one cites Windows 10 as the minimum. Since Northstar requires EA auth with a running client, that could spell the end of Win7 feasibility, unless Maxima would happen to support it for some reason... (I don't think an LSX proxy on another PC is something most people would run alternatively) 

I'll have to ask about Maxima and maybe someone with a VM could verify how EA App fares currently there

EDIT: Maxima is written in Rust and that apparently doesn't support Win7 in its main Windows target, so that'd be a no there, unless someone managed to compile it with the legacy win7 tier3 target and it happened to work...
